migrations/20240110180056_set_idle_in_transaction_timeout.sql

```diff
@@ -0,0 +1,2 @@
+-- If running worker in transactional mode, this ensures we clean up any open transactions.
+SET idle_in_transaction_session_timeout='2min';
```

Comment:
I believe this is a client connection setting. I think we want to set it on every connection we open?

But maybe this setting sets a decent default?

https://docs.rs/sqlx-core/0.7.3/src/sqlx_core/pool/options.rs.html#147

We could override it when we make `PoolOptions` if we want to change it.

Reply:
I saw that one, but that `idle_timeout` refers to a connection idle timeout enforced by (I believe) sqlx, at least that's what I gather looking at the docs: https://docs.rs/sqlx/latest/sqlx/pool/struct.PoolOptions.html#method.idle_timeout

Although valuable to set that one too, the reason to have a tx timeout enforced by PG is to have PG clean up any transactions left open after a pod crashes or becomes unresponsive for whatever other reason.

> But if we want to change it on the cluster, I'm struggling to find proof that this will do it. I thought this was a connection level setting? Does putting this in a migration really change the DB/cluster default?

I can update the query to make it affect the current user. As long as the same user is used to run migrations and the services, we'll be fine. Ideally this should be updated in the postgres configuration though.